pandas/core/groupby/generic.py

```diff
@@ -917,8 +917,11 @@ def transform(self, func, *args, **kwargs):
             s = klass(res, indexer)
             results.append(s)
 
-        from pandas.core.reshape.concat import concat
-        result = concat(results).sort_index()
+        if results:
```

Comment:
pls add a 1-line comment here on why you are doing this

Reply:
Done.